client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -0,0 +1,147 @@
+ */
+import { installPlugin } from 'state/plugins/installed/actions';
+import { fetchPluginData } from 'state/plugins/wporg/actions';
+import { getPlugin } from 'state/plugins/wporg/selectors';
+import { getPlugins } from 'state/plugins/installed/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import ProgressBar from 'components/progress-bar';
+import QueryPluginKeys from 'components/data/query-plugin-keys';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import SetupHeader from './setup-header';
+import { setFinishedInstallOfRequiredPlugins } from 'woocommerce/state/sites/setup-choices/actions';
+
+const requiredPlugins = [
+	'woocommerce',
+	// 'wc-api-dev',
+	'woocommerce-gateway-stripe',
+	'woocommerce-services',
+];
+
+class RequiredPluginsInstallView extends Component {
+	static propTypes = {
+		fetchPluginData: PropTypes.func.isRequired,
+		installPlugin: PropTypes.func.isRequired,
+		plugins: PropTypes.array,
+		setFinishedInstallOfRequiredPlugins: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+		} ),
+		wpOrg: PropTypes.array,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			installingPlugin: null,
+			progress: 0,
+		};
+	}
+
+	componentDidMount = () => {
+		const { plugins, site } = this.props;
+
+		if ( site && plugins && plugins.length ) {
+			this.installPlugins( plugins );
+		}
+
+		this.getWporgPluginData();
+	}
+
+	componentDidUpdate = ( prevProps ) => {
+		const { plugins, site } = this.props.plugins;
+		if (
+			( site && plugins && plugins.length && ! this.state.installingPlugin ) ||
+			( prevProps.plugins && plugins.length > prevProps.plugins.length )
+		) {
+			this.installPlugins( this.props.plugins );
+		}
+	}
+
+	getWporgPluginData() {
+		requiredPlugins.map( plugin => {
+			const pluginData = getPlugin( this.props.wporg, plugin );
+			if ( ! pluginData ) {
+				this.props.fetchPluginData( plugin );
+			}
+		} );
+	}
+
+	installPlugins = ( plugins ) => {
+		const { site, wporg } = this.props;
+		let isRunningInstall = false;
+		for ( let i = 0; i < requiredPlugins.length; i++ ) {
+			const slug = requiredPlugins[ i ];
+			const plugin = find( plugins, { slug } );
+			if ( ! plugin ) {
+				if ( ! wporg[ slug ] ) {
+					return;
+				}
+				const wporgPlugin = getPlugin( wporg, slug );
+				this.setState( {
+					installingPlugin: slug,
+					progress: this.state.progress + 25
+				} );
+				this.props.installPlugin( site.ID, wporgPlugin );
+				isRunningInstall = true;
+				return;
+			}
+		}
+		if ( ! isRunningInstall ) {
+			this.props.setFinishedInstallOfRequiredPlugins(
+				site.ID,
+				true
+			);
+		}
+	}
+
+	render = () => {
+		const { translate, site } = this.props;
+		return (
+			<div className="card dashboard__setup-wrapper">
+				{ site && <QueryJetpackPlugins siteIds={ [ site.ID ] } /> }
+				{ site.canUpdateFiles && <QueryPluginKeys siteId={ site.ID } /> }
```

Comment:
What is this used for? I'm not familiar with `QueryPluginKeys`, but best I can tell, it has something to do with premium plugins.